mypy 17.0.0 errors

When upgrading to mypy version 17.0.0 the following errors occur:

```
$ make mypy
python3 -m mypy arkouda
arkouda/numpy/numeric.py:2607: error: Incompatible return value type (got "signedinteger[_8Bit] | signedinteger[_16Bit] | signedinteger[_32Bit] | signedinteger[_64Bit]", expected "signedinteger[_64Bit]")  [return-value]
arkouda/numpy/numeric.py:2612: error: Incompatible return value type (got "signedinteger[_8Bit] | signedinteger[_16Bit] | signedinteger[_32Bit] | signedinteger[_64Bit]", expected "signedinteger[_64Bit]")  [return-value]
arkouda/numpy/numeric.py:2617: error: Incompatible return value type (got "signedinteger[_8Bit] | signedinteger[_16Bit] | signedinteger[_32Bit] | signedinteger[_64Bit]", expected "signedinteger[_64Bit]")  [return-value]
arkouda/numpy/util.py:907: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
arkouda/numpy/segarray.py:72: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
Found 3 errors in 1 file (checked 109 source files)
make: *** [Makefile:762: mypy] Error 1


```

--- a/arkouda/numpy/numeric.py
+++ b/arkouda/numpy/numeric.py
@@ -2596,25 +2596,29 @@ def count_nonzero(pda: pdarray) -> np.int64:
 
     """
 
+    from arkouda.numpy.dtypes import can_cast
     from arkouda.numpy.util import is_numeric
 
     #  Handle different data types.
 
     if is_numeric(pda):
         value = sum((pda != 0).astype(np.int64))
-        if not isinstance(value, np.int64):
+        if can_cast(value, np.int64):
+            return np.int64(value)
+        else:
             raise ValueError("summing the pdarray did not generate a scalar")
-        return value
     elif pda.dtype == bool:
         value = sum((pda).astype(np.int64))
-        if not isinstance(value, np.int64):
+        if can_cast(value, np.int64):
+            return np.int64(value)
+        else:
             raise ValueError("summing the pdarray did not generate a scalar")
-        return value
     elif pda.dtype == str:
         value = sum((pda != "").astype(np.int64))
-        if not isinstance(value, np.int64):
+        if can_cast(value, np.int64):
+            return np.int64(value)
+        else:
             raise ValueError("summing the pdarray did not generate a scalar")
-        return value
     raise TypeError("pda must be numeric, bool, or str pdarray")
 
 

